motrainer/motrainer.py

```diff
@@ -0,0 +1,117 @@
+import warnings
+import xarray as xr
+import numpy as np
+import dask.bag as db
+
+MOT_DIMS = ["space", "time"]  # Expected xr.Dataset dimensions
+
+
+@xr.register_dataset_accessor("mot")
+class MOTrainerDataset:
+    """Xarray.Dataset extension for MOTrainer
+
+    It contains methods facilitating distributed training tasks. 
+
+    Following constraints are applied to a regular Dataset:
+        1. The Dataset should have exactly two dimensions: "space" and "time"
+        2. There are no duplicated entries in coordinates
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj
+
+    def is_valid(self) -> bool:
+        """Check if a Dataset is valid for MOTrainer.
+        The following checks will be applied:
+            - The Dastaset has exactly 2 dimensions
+            - The 2 dims are "space" and "time"
+            - There are no duplicated coordinates
+        A UserWarning will be raised for each failed check.
+
+        Returns:
+            bool: Result of check in Boolean.
+                  If all checks are passed, it will be True. Otherwise False.
+        """
+
+        flag_valid = True
+
+        # Dimension size should be 2
+        if len(self._obj.dims) != 2:
+            warnings.warn(
+                f'Dataset should have two dimensions: "space" and "time"', UserWarning
+            )
+            flag_valid = False
+
+        # space and time dimensions should exist
+        for dim in MOT_DIMS:
+            if not (dim in self._obj.dims):
+                warnings.warn(f"{dim} not found in the dimensions", UserWarning)
+                flag_valid = False
+
+        # Check duplicated coordinates
+        for coord in self._obj.coords:
+            if np.unique(self._obj[coord]).shape != self._obj[coord].shape:
+                warnings.warn(f"Duplicated coordinates found in {coord}", UserWarning)
+                flag_valid = False
+
+        return flag_valid
+
+    def model_split(self, identifier: dict | str):
+        """Split a Dataset by indentifier for independent training tasks.
```

Comment:
I think it is confusing if the function name is `model_split` but the documentation says `"Split a Dataset"`. Can we rename the function to `dataset_split`? Also, can we say `Xarray Dataset` to make it more explicit?